x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -144,7 +191,11 @@ private static Match readFrom(StreamInput in) throws IOException {
         if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_QUERY_BUILDER_IN_SEARCH_FUNCTIONS)) {
             queryBuilder = in.readOptionalNamedWriteable(QueryBuilder.class);
         }
-        return new Match(source, field, query, queryBuilder);
+        Expression options = null;
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_MATCH_OPTIONS)) {
+            options = in.readOptionalNamedWriteable(Expression.class);
```

Comment:
a quick note - technically serializing options should not be needed - because by the time these are sent to the data node, `Match` objects should already contain a `queryBuilder` that's already serialized, that already uses the options.

Reply:
Good catch - removed in d9c62d351d3216f8979f35168d362681372f02c5